### PR TITLE
ARACHNE-2534. Added ability to retrieve all StudyDataSourceLinks from repository

### PR DESCRIPTION
**Pull request by aklochkova**

### CODE REVIEW

File: src/main/java/com/odysseusinc/arachne/portal/repository/StudyDataSourceLinkRepository.java [hunk 1]
```diff
@@ -58,7 +58,9 @@
     StudyDataSourceLink findByDataSourceIdAndStudyId(@Param("dataSourceId") Long dataSourceId,
                                                      @Param("studyId") Long studyId);
 
-    List<StudyDataSourceLink> findByStudyId(Long studyId);
+    //Have to do native query to ignore @Where in StudyDataSourceLink entity
+    @Query(nativeQuery = true, value = "SELECT * FROM studies_data_sources l WHERE l.study_id =:studyId")
+    List<StudyDataSourceLink> findByStudyId(@Param("studyId") Long studyId);
```
Comment (pavgra) [inline src/main/java/com/odysseusinc/arachne/portal/repository/StudyDataSourceLinkRepository.java]: State clearly that the query results are going to include deleted items

File: src/main/java/com/odysseusinc/arachne/portal/api/v1/dto/converters/study/BaseUserStudyItemToStudyDTOConverter.java [hunk 2]
```diff
@@ -37,17 +42,21 @@
...
     public S convert(AbstractUserStudyListItem source) {
 
+        final Study study = source.getStudy();
         S studyDto = createResultObject();
 
-        StudyDTO baseObject = conversionService.convert(source.getStudy(), getDtoClass());
+        StudyDTO baseObject = conversionService.convert(study, getDtoClass());
         baseObject.setFavourite(source.getFavourite());
 
+        List<StudyDataSourceLink> links = study.getDataSources().stream().filter(ds -> DataSourceStatus.DELETED.equals(ds.getStatus())).collect(Collectors.toList());
```
Comment (pavgra) [inline src/main/java/com/odysseusinc/arachne/portal/api/v1/dto/converters/study/BaseUserStudyItemToStudyDTOConverter.java]: What is this logic for?

Comment (aklochkova) [reply src/main/java/com/odysseusinc/arachne/portal/api/v1/dto/converters/study/BaseUserStudyItemToStudyDTOConverter.java]: When Study is being converted to StudyDTO in BaseStudyToCommonStudyDTOConverter its DataSourceLinks are get from repository via non-nativeQuery method StudyDataSourceLinkRepository#findByStudyId(Long, EntityGraph) and are filtered then by @Where in StudyDataSourceLink. So I get DataSourceLinks from incoming study (there are "deleted" among them) and convert them here.

Comment (pavgra) [reply src/main/java/com/odysseusinc/arachne/portal/api/v1/dto/converters/study/BaseUserStudyItemToStudyDTOConverter.java]: So `baseObject.setDataSources(dataSourceDTOS);` will set only _deleted_ sources in the resulting DTO. W/o still active DSs. Which is not appropriate behavior

File: src/main/java/com/odysseusinc/arachne/portal/repository/StudyDataSourceLinkRepository.java [hunk 3]
```diff
@@ -58,7 +58,9 @@
     StudyDataSourceLink findByDataSourceIdAndStudyId(@Param("dataSourceId") Long dataSourceId,
                                                      @Param("studyId") Long studyId);
 
-    List<StudyDataSourceLink> findByStudyId(Long studyId);
+    //Have to do native query to include items with "deleted_at" != NULL
+    @Query(nativeQuery = true, value = "SELECT * FROM studies_data_sources l WHERE l.study_id =:studyId")
+    List<StudyDataSourceLink> findByStudyId(@Param("studyId") Long studyId);
```
Comment (pavgra) [inline src/main/java/com/odysseusinc/arachne/portal/repository/StudyDataSourceLinkRepository.java]: naming. to clearly show that the method returns deleted items

**Pull request merged**